geem/static/geem/js/geem_validation.js

```diff
@@ -72,11 +67,44 @@ function update_user_grid(grid_options, data) {
 /**
  * Update ontology grid headers.
  * Must be called in ``portal_entity_form_callback``.
- * @param grid_options - Ontology validation grid options
+ * @param {Object} grid_options - Ontology validation grid options
  */
 function update_ontology_grid(grid_options) {
 	const column_defs = top.form.components.map(function(component) {
 		return {headerName: component.label, field: component.id}
-	})
+	});
 	grid_options.api.setColumnDefs(column_defs)
 }
+
+
+/**
+ * Download data from a grid instance.
+ * @param {Object} grid_options - Grid options of grid to download from
+ * @param {('text/csv'|'text/tab-separated-values')} file_type - MIME
+ * 	type of file to download
+ */
+function download_grid(grid_options, file_type) {
+	let data_str = grid_options.api.getDataAsCsv();
+	let file_name;
+
+	if (file_type === 'text/tab-separated-values') {
+		data_str = data_str.replace(/,/g, '\t');
```

Comment:
Issue is comma separated data, if it has a text field, usually with quotations in it "....", needs to have quotations removed, as I recall.

Reply:
Is there a way to have whole system work with tab separated values by default and write out .csv if necessary?
So grid_options.api.getDataAsTsv();

Best to wean people off csv.

Reply:
> Issue is comma separated data, if it has a text field, usually with quotations in it "....", needs to have quotations removed, as I recall.

I don't 100% understand.

The data is like this:

`"\"SampleId\",\"Sample\r\"\r\n\"small_simple1\",\"Chicken Breast\r\"\r\n\"small_simple2\",\"Baked Potato\r\"\r\n\"small_simple3\",\"Canned Corn\r\"\r\n\"small_simple4\",\"Frozen Yogurt\r\"\r\n\"small_simple5\",\"Apple Pie\r\"\r\n\"\",\"\"\r\n"`

There are quotations everywhere, but they do not show up in the downloaded file:

![image](https://user-images.githubusercontent.com/34198261/70366435-2f65c000-184c-11ea-9daa-ad7328f031d7.png)


> Is there a way to have whole system work with tab separated values by default and write out .csv if necessary?
So grid_options.api.getDataAsTsv();

[Unfortunately, no.](https://www.ag-grid.com/javascript-grid-export/) :frowning_face:

Reply:
look at the downloaded "tabular" file in a text editor (not spreadsheet app) to verify if the quotes are making it in there.  If you are making a tabular file ONLY by replacing commas with tabs, then I'm pretty sure the quotes are hanging around as an artifact.

Reply:
There are indeed quotations hanging around. They are also present in the ``.csv`` file.

Reply:
The .csv file uses quotation marks, at a minimum around any text that has spaces or commas in it. It may even put them around numbers and single words, I'm not sure, as part of a standard. As well, deliberate quotes inside text must be escaped.
But a .tsv file should NOT have quotes around its textual values unless they were deliberately part of the text, e.g. a field with a string of text: ' he said "WHAT!" very loudly'. Tsv files don't need escaped quotes. That's where the two formats really stray from each other.
Here is a web example of conversion challenge:
Name,Age
Bob,12
"Doe, John",13
"William ""Billy"" Bob",14
could need to be properly transformed to
Name\tAge
Bob\t12
Doe, John\t13
William "Billy" Bob\t14

Another ref: https://unix.stackexchange.com/questions/359832/converting-csv-to-tsv

It seems like one could try a round trip.  csv to json to tsv perhaps. I don't see any pure javascript csv to tsv conversion code, arg.